### DISCUSSION TITLE
DataTable Support Fixed Columns when a lot of Columns?

### DISCUSSION THREAD

**Discussion by dybxin** [carbon-design-system/carbon-components-svelte#1736] [Q&A]
DataTable support fixed columns Like the following picture show:
![image](https://github.com/carbon-design-system/carbon-components-svelte/assets/14888437/eb3d666f-7bb5-413f-b28b-4fccc493c169)

or Having a built-in way for tables can be scroll in x (or even in y)?

**Comment by theetrain** [comment]
There currently isn't a built-in way to achieve this, but you may use CSS overrides such as this: https://github.com/carbon-design-system/carbon-components-svelte/discussions/1512#discussioncomment-3971540

You may advocate for sticky columns in the Carbon core repository before it can make its way to carbon-components-svelte https://github.com/carbon-design-system/carbon/issues

**Comment by dybxin** [comment]
@theetrain  advocate datatable support sticky columns, [the official reply](https://github.com/carbon-design-system/carbon/issues/13874).

there is a library [ibm-products repository](https://github.com/carbon-design-system/ibm-products) provided by the [carbon-design-system](https://github.com/carbon-design-system). contains Datagrid components [demo here](https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-datagrid-datagrid-canary--frozen-columns), well support current needs, but unfortunately  [currently only support react](https://github.com/carbon-design-system/ibm-products/issues/2560).

**Reply by theetrain** [reply]
Unfortunately since it's not supported by `@carbon/react` it's unlikely we'll port that functionality to `carbon-components-svelte` since we have other priorities you may review in https://github.com/carbon-design-system/carbon-components-svelte/discussions/1614

Though this is achievable with CSS overrides like in my shared example. There is also sticky header support today: https://carbon-components-svelte.onrender.com/components/DataTable#sticky-header
